orchestrator/modules/operators/_cleanup.py

```diff
@@ -2,56 +2,50 @@
 # SPDX-License-Identifier: MIT
 
 import logging
-import signal
 import typing
+from collections import OrderedDict
+from collections.abc import Callable
 
 import ray
 from ray.actor import ActorHandle
 
-shutdown = False
+shutdown_signal_received = False
 CLEANER_ACTOR = "resource_cleaner"
 
 moduleLog = logging.getLogger("orchestration_cleanup")
+signal_cleanup_callbacks: dict[str, Callable[[], None]] = OrderedDict()
```

Comment:
I'd call these `cleanup_callback_functions` so that it's clear that they're functions